### DISCUSSION TITLE
validate Method

### DISCUSSION THREAD

**Discussion by kevinswarner** [dynamoose/dynamoose#1229] [Q&A]
Is there any way to run validation on a document directly instead of relying on validation on an attempted read or write operation? For example...

```js
const Widget = dynamoose.model('Widget', {
  email: {
    type: String,
    required: true,
    validate: /SOME_REGEX_HERE/
  }
});

const widget = new Widget({ email: 'foo@bar.com' });

const validationResults = await widget.validate();
```

I am not seeing any methods in the documentation for this kind of thing. Just wanted to check in case I am missing something. Thanks.

**Comment by fishcharlie** [comment]
@kevinswarner Nope. All of this behavior is internal currently. I'd suggest rolling your own system or using another npm package for this.

Feel free to submit a feature request for this, but I see it being pretty low priority currently.
